### ISSUE TITLE
Delphi 2007 support

### ISSUE THREAD

**Issue by RrnR** [pavkam/tzdb#47]
I have several Delphi 2007 projects that I want to use TZDB for.  It ticks a lot of boxes and I appreciate the effort that must go into maintaining it. ;-) ) For historical reasons, I'm stuck with D2007.  The TZDB source set contains 2 files called TZDB.pas.  One is small (tens of K), one larger (100's of k).  

If I set up my library paths to make the smaller one visible, the applications compile/build.  The build seems to produce a fresh TZDB.dcu 

If I set up my library paths to make the larger one visible, the compile snags, announcing that I need a later version of Delphi to compile. 

I'm feeling slightly vulnerable here when it comes to updating TZDB.  I read that all you need to do to use TZDB is put TZDB.pas on your library path and `use tzdb` in your `uses` clause.  No mention of TZDB.inc but it must be a requirement also.

Is it intended to maintain Delphi 2007 support at this level? I.e. if I update TZDB.pas (the lesser) and TZDB.inc every so often I'll be OK?

Also I'm just wondering: Why are there two different source files with the same name in a source set?

**Comment by pavkam** [comment]
Hi,

TZDB requires Delphi XE+ to work. Or a version of FreePascal. You're
probably using the source version (smaller one) instead of dist version.

I am not sure why the source version works for Delphi 7. Should have
complained as well. But if it compiler at works than you van use it as
such. Just make sure it actually works.

Updating the inc file only is OK in that case.bthe dist version puts
everything into one big TZDB pas rile (all includes and such)

I guess TZDB source version does not do the version checks so that's why it
could compile. Though I'm very unsure why (too old for me to remember
what's miswint) maybe it thinks it's FreePascal or something.

In short, use then source version (smaller one) and test it works for you.
If it does, use it.

Regards,

*Alexandru Ciobanu*
Web: https://alex.ciobanu.org/ <http://alex.ciobanu.org/>
Email: ***@***.***

**Comment by pavkam** [comment]
Closing request as it won't be implemented. Not planning to support old Delphi versions.
